In [2]:
import pandas as pd
import numpy as np

# Load Dataset

In [11]:
df = pd.read_csv(r"C:\Users\Suddala Vaishnavi\Downloads\mental health condition\project1.csv")

In [12]:
df

Unnamed: 0      id     Name  Gender   Age           City  \
0               0  140700   Shivam    Male  53.0  Visakhapatnam   
1               1  140701    Sanya  Female  58.0        Kolkata   
2               2  140702     Yash    Male  53.0         Jaipur   
3               3  140703   Nalini  Female  23.0         Rajkot   
4               4  140704  Shaurya    Male  47.0         Kalyan   
...           ...     ...      ...     ...   ...            ...   
93795       93795  234495     Zoya  Female  49.0         Jaipur   
93796       93796  234496    Shlok    Male  29.0      Ahmedabad   
93797       93797  234497    Rishi    Male  24.0  Visakhapatnam   
93798       93798  234498   Eshita  Female  23.0         Kalyan   
93799       93799  234499    Gauri  Female  43.0       Varanasi   

      Working Professional or Student              Profession  Work Pressure  \
0                Working Professional                   Judge            2.0   
1                Working Professional  Educational Consultant            2.0   
2                Working Professional                 Teacher            4.0   
3                             Student                     NaN            NaN   
4                Working Professional                 Teacher            5.0   
...                               ...                     ...            ...   
93795            Working Professional                   Pilot            3.0   
93796            Working Professional                   Pilot            5.0   
93797                         Student                     NaN            NaN   
93798            Working Professional       Marketing Manager            4.0   
93799            Working Professional  Educational Consultant            5.0   

       Job Satisfaction     Sleep Duration Dietary Habits  Degree  \
0                   5.0  Less than 5 hours       Moderate     LLB   
1                   4.0  Less than 5 hours       Moderate    B.Ed   
2                   1.0          7-8 hours       Moderate  B.Arch   
3                   NaN  More than 8 hours       Moderate     BSc   
4                   5.0          7-8 hours       Moderate     BCA   
...                 ...                ...            ...     ...   
93795               5.0  Less than 5 hours       Moderate     BSc   
93796               1.0          7-8 hours       Moderate      BE   
93797               NaN          7-8 hours       Moderate  B.Tech   
93798               2.0          5-6 hours        Healthy      BA   
93799               2.0  More than 8 hours        Healthy    B.Ed   

      Suicidal thoughts  Work/Study Hours  Financial Stress  \
0                    No               9.0               3.0   
1                    No               6.0               4.0   
2                   Yes              12.0               4.0   
3                   Yes              10.0               4.0   
4                   Yes               3.0               4.0   
...                 ...               ...               ...   
93795               Yes               2.0               2.0   
93796               Yes              11.0               3.0   
93797                No               7.0               1.0   
93798               Yes               7.0               5.0   
93799                No              11.0               2.0   

      Family History of Mental Illness  
0                                  Yes  
1                                   No  
2                                   No  
3                                   No  
4                                   No  
...                                ...  
93795                              Yes  
93796                              Yes  
93797                               No  
93798                              Yes  
93799                               No  

[93800 rows x 17 columns]

In [13]:
df.drop(["Unnamed: 0"],axis=1,inplace=True)

In [14]:
df

id     Name  Gender   Age           City  \
0      140700   Shivam    Male  53.0  Visakhapatnam   
1      140701    Sanya  Female  58.0        Kolkata   
2      140702     Yash    Male  53.0         Jaipur   
3      140703   Nalini  Female  23.0         Rajkot   
4      140704  Shaurya    Male  47.0         Kalyan   
...       ...      ...     ...   ...            ...   
93795  234495     Zoya  Female  49.0         Jaipur   
93796  234496    Shlok    Male  29.0      Ahmedabad   
93797  234497    Rishi    Male  24.0  Visakhapatnam   
93798  234498   Eshita  Female  23.0         Kalyan   
93799  234499    Gauri  Female  43.0       Varanasi   

      Working Professional or Student              Profession  Work Pressure  \
0                Working Professional                   Judge            2.0   
1                Working Professional  Educational Consultant            2.0   
2                Working Professional                 Teacher            4.0   
3                             Student                     NaN            NaN   
4                Working Professional                 Teacher            5.0   
...                               ...                     ...            ...   
93795            Working Professional                   Pilot            3.0   
93796            Working Professional                   Pilot            5.0   
93797                         Student                     NaN            NaN   
93798            Working Professional       Marketing Manager            4.0   
93799            Working Professional  Educational Consultant            5.0   

       Job Satisfaction     Sleep Duration Dietary Habits  Degree  \
0                   5.0  Less than 5 hours       Moderate     LLB   
1                   4.0  Less than 5 hours       Moderate    B.Ed   
2                   1.0          7-8 hours       Moderate  B.Arch   
3                   NaN  More than 8 hours       Moderate     BSc   
4                   5.0          7-8 hours       Moderate     BCA   
...                 ...                ...            ...     ...   
93795               5.0  Less than 5 hours       Moderate     BSc   
93796               1.0          7-8 hours       Moderate      BE   
93797               NaN          7-8 hours       Moderate  B.Tech   
93798               2.0          5-6 hours        Healthy      BA   
93799               2.0  More than 8 hours        Healthy    B.Ed   

      Suicidal thoughts  Work/Study Hours  Financial Stress  \
0                    No               9.0               3.0   
1                    No               6.0               4.0   
2                   Yes              12.0               4.0   
3                   Yes              10.0               4.0   
4                   Yes               3.0               4.0   
...                 ...               ...               ...   
93795               Yes               2.0               2.0   
93796               Yes              11.0               3.0   
93797                No               7.0               1.0   
93798               Yes               7.0               5.0   
93799                No              11.0               2.0   

      Family History of Mental Illness  
0                                  Yes  
1                                   No  
2                                   No  
3                                   No  
4                                   No  
...                                ...  
93795                              Yes  
93796                              Yes  
93797                               No  
93798                              Yes  
93799                               No  

[93800 rows x 16 columns]

# Observation:
- Dataset has been loaded succesfully using pd.read_csv()

# Understanding the structure of the Dataset

In [15]:
df.shape

(93800, 16)

# Obesrvation:
- The dataset contains 93800 rows and 16 columns which representing  students mental health conditions.

# Previewing the first few rows

In [16]:
df.head()

id     Name  Gender   Age           City  \
0  140700   Shivam    Male  53.0  Visakhapatnam   
1  140701    Sanya  Female  58.0        Kolkata   
2  140702     Yash    Male  53.0         Jaipur   
3  140703   Nalini  Female  23.0         Rajkot   
4  140704  Shaurya    Male  47.0         Kalyan   

  Working Professional or Student              Profession  Work Pressure  \
0            Working Professional                   Judge            2.0   
1            Working Professional  Educational Consultant            2.0   
2            Working Professional                 Teacher            4.0   
3                         Student                     NaN            NaN   
4            Working Professional                 Teacher            5.0   

   Job Satisfaction     Sleep Duration Dietary Habits  Degree  \
0               5.0  Less than 5 hours       Moderate     LLB   
1               4.0  Less than 5 hours       Moderate    B.Ed   
2               1.0          7-8 hours       Moderate  B.Arch   
3               NaN  More than 8 hours       Moderate     BSc   
4               5.0          7-8 hours       Moderate     BCA   

  Suicidal thoughts  Work/Study Hours  Financial Stress  \
0                No               9.0               3.0   
1                No               6.0               4.0   
2               Yes              12.0               4.0   
3               Yes              10.0               4.0   
4               Yes               3.0               4.0   

  Family History of Mental Illness  
0                              Yes  
1                               No  
2                               No  
3                               No  
4                               No

# Observation:
- The preview shows the first few rows of the data and helps us check if the file loaded correctly.

# Dataset Structure and Data Types

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93800 entries, 0 to 93799
Data columns (total 16 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   id                                93800 non-null  int64  
 1   Name                              93800 non-null  object 
 2   Gender                            93800 non-null  object 
 3   Age                               93800 non-null  float64
 4   City                              93800 non-null  object 
 5   Working Professional or Student   93800 non-null  object 
 6   Profession                        69168 non-null  object 
 7   Work Pressure                     75022 non-null  float64
 8   Job Satisfaction                  75026 non-null  float64
 9   Sleep Duration                    93800 non-null  object 
 10  Dietary Habits                    93795 non-null  object 
 11  Degree                            93798 non-null  object 
 12  Suic

# Observation:
- The dataset contains 93800 records and 16 columns.
- Out of the 19 columns
          - 6 are numerical (int64/float64).
          - 10 are categorical(object type).
- The dataset occupies approximatly 11.6

# Statistical Overview of Numerical Data

In [18]:
df.describe()

id           Age  Work Pressure  Job Satisfaction  \
count   93800.000000  93800.000000   75022.000000       75026.00000   
mean   187599.500000     40.321685       3.011797           2.96092   
std     27077.871962     12.393480       1.403563           1.41071   
min    140700.000000     18.000000       1.000000           1.00000   
25%    164149.750000     29.000000       2.000000           2.00000   
50%    187599.500000     42.000000       3.000000           3.00000   
75%    211049.250000     51.000000       4.000000           4.00000   
max    234499.000000     60.000000       5.000000           5.00000   

       Work/Study Hours  Financial Stress  
count      93800.000000      93800.000000  
mean           6.247335          2.978763  
std            3.858191          1.414604  
min            0.000000          1.000000  
25%            3.000000          2.000000  
50%            6.000000          3.000000  
75%           10.000000          4.000000  
max           12.000000          5.000000

# Obeservation:
- The summary shows fundamental measures like mean,min,max, and standard deviation for all numerical columns.                                                                                  

# Statistical Overview of categorical Data

In [20]:
df.describe(include=("O"))

Name Gender    City Working Professional or Student Profession  \
count   93800  93800   93800                           93800      69168   
unique    374      2      68                               2         64   
top     Rohan   Male  Kalyan            Working Professional    Teacher   
freq     2112  51262    4387                           75028      16385   

           Sleep Duration Dietary Habits    Degree Suicidal thoughts  \
count               93800          93795     93798             93800   
unique                 31             22        87                 2   
top     Less than 5 hours       Moderate  Class 12                No   
freq                25661          33018      9812             47375   

       Family History of Mental Illness  
count                             93800  
unique                                2  
top                                  No  
freq                              47118

# Observation:
- The summary provides frequency,counts,top and number of unique entries.

# Null Value Count Across All Columns

In [21]:
df.isnull().sum()

id                                      0
Name                                    0
Gender                                  0
Age                                     0
City                                    0
Working Professional or Student         0
Profession                          24632
Work Pressure                       18778
Job Satisfaction                    18774
Sleep Duration                          0
Dietary Habits                          5
Degree                                  2
Suicidal thoughts                       0
Work/Study Hours                        0
Financial Stress                        0
Family History of Mental Illness        0
dtype: int64

# Percentage of Missing Values

In [22]:
round(df.isnull().sum()/len(df)*100)

id                                   0.0
Name                                 0.0
Gender                               0.0
Age                                  0.0
City                                 0.0
Working Professional or Student      0.0
Profession                          26.0
Work Pressure                       20.0
Job Satisfaction                    20.0
Sleep Duration                       0.0
Dietary Habits                       0.0
Degree                               0.0
Suicidal thoughts                    0.0
Work/Study Hours                     0.0
Financial Stress                     0.0
Family History of Mental Illness     0.0
dtype: float64

# Observation:
- 6 columns containg missing values:
- Profession --> 26% missing values.              
- Work Pressure --> 20% missing values.                              
- Job Satisfaction --> 20% missing values.

In [23]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93800 entries, 0 to 93799
Data columns (total 16 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   id                                93800 non-null  int64  
 1   Name                              93800 non-null  object 
 2   Gender                            93800 non-null  object 
 3   Age                               93800 non-null  float64
 4   City                              93800 non-null  object 
 5   Working Professional or Student   93800 non-null  object 
 6   Profession                        69168 non-null  object 
 7   Work Pressure                     75022 non-null  float64
 8   Job Satisfaction                  75026 non-null  float64
 9   Sleep Duration                    93800 non-null  object 
 10  Dietary Habits                    93795 non-null  object 
 11  Degree                            93798 non-null  object 
 12  Suic

# Value Count Summary(Mental Health of a students and working professionals):

## 1.id
- It contain integers values.
- Each Id contain different integer id number.
- fully complete,no missing values.

## 2.Name
- name of a person.
- no missing values.

## 3.Gender
- It contain gender of a person(Male/Female).
- no missing values.

## 4.Age
- it contain age of a person in numerical values.
- no missing values.

## 5. Working Professional or Student
- it contain a person is  Working Professional or Student.
- no missing values.

## 6. Profession   
- it contain, a person is in which profession(teacher,loyer,etc..).
- 24632 missing values

## 7.Work Pressure 
- the work pressure of a person in daily life.
- 18774 missing values.

## 8.Job Satisfaction
- Rating of satisfaction with their job.
- 18774 missing values.

## 9.Sleep Duration
- Average sleep duration(like 6 hours,7 hours).
- 5 missing values

## 10. Dietary Habits.
- eating habits.
- 2 missing values.

## 11.Degree
- The Degree/course they are pursing.
- 2 missing values.

## 12.suicidal thoughts
- the thougths of a persons.
- no missing values.

## 13.Work/Study Hours
- Average number of hours spent working/studying per day.

## 14. Finacial Stress
- Numeric rating of financial pressure(like difficulty affording tution,expensenses.

## 15.city
- city of residence.

## 16.Family History ofMental illness.
- whether any close family members have mental illness or not.

In [31]:
df["id"].value_counts()

id
140700    1
203240    1
203238    1
203237    1
203236    1
         ..
171965    1
171964    1
171963    1
171962    1
234499    1
Name: count, Length: 93800, dtype: int64

# Observation
- every id occurs 1 time in id column, so it shows 1 for each id.
- and also it shows its length that is 93800 and also dtype(datatype) is int64.

In [35]:
df["Age"].value_counts()

Age
56.00    3432
49.00    3425
38.00    3008
53.00    2958
57.00    2875
47.00    2740
46.00    2728
18.00    2614
59.00    2551
54.00    2549
51.00    2544
43.00    2506
48.00    2459
20.00    2335
45.00    2329
50.00    2311
24.00    2296
42.00    2295
28.00    2292
39.00    2279
44.00    2242
29.00    2152
41.00    2093
33.00    2057
58.00    1982
25.00    1953
40.00    1950
35.00    1940
21.00    1930
55.00    1923
23.00    1870
37.00    1789
34.00    1785
36.00    1742
19.00    1731
27.00    1719
60.00    1701
52.00    1687
31.00    1660
22.00    1417
26.00    1362
30.00    1361
32.00    1227
32.08       1
Name: count, dtype: int64

In [36]:
len(df["Age"].value_counts())

44

# Observation
- It shows Ages of a persons and how many times each age repeated in age cloumns.
- and ages contain float values.
- and also it shows dtype(datatype) is int64.
- and its length is 44.

In [38]:
df["Work Pressure"].value_counts()

Work Pressure
2.0    16255
5.0    15113
4.0    14974
3.0    14650
1.0    14030
Name: count, dtype: int64

# Observation
- It shows the work pressure of a persons.
- it contain ratings out of 5.0.
- it is in a float values.

In [40]:
df["Job Satisfaction"].value_counts()

Job Satisfaction
2.0    16971
5.0    14827
1.0    14793
3.0    14464
4.0    13971
Name: count, dtype: int64

# Observation
- It shows the job satisfaction of a persons.
- it contain rating out of 5.0.
- it is in float values.


In [41]:
df["Work/Study Hours"].value_counts()

Work/Study Hours
10.0    9450
11.0    8555
9.0     8315
0.0     8132
12.0    7657
2.0     7083
6.0     7029
7.0     6646
1.0     6525
3.0     6261
5.0     6118
4.0     6079
8.0     5950
Name: count, dtype: int64

# Observation
- It shows the work/study hours of a person.
- it contain how many hours in a day a person work/studied.
- The hours contain float values.


In [42]:
df["Financial Stress"].value_counts()

Financial Stress
2.0    21151
5.0    18694
4.0    18453
1.0    18341
3.0    17161
Name: count, dtype: int64

# Observation
- It shows financial stress of a person.
- it contain rating out of 5.0.
- it contain float values.

In [43]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93800 entries, 0 to 93799
Data columns (total 16 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   id                                93800 non-null  int64  
 1   Name                              93800 non-null  object 
 2   Gender                            93800 non-null  object 
 3   Age                               93800 non-null  float64
 4   City                              93800 non-null  object 
 5   Working Professional or Student   93800 non-null  object 
 6   Profession                        69168 non-null  object 
 7   Work Pressure                     75022 non-null  float64
 8   Job Satisfaction                  75026 non-null  float64
 9   Sleep Duration                    93800 non-null  object 
 10  Dietary Habits                    93795 non-null  object 
 11  Degree                            93798 non-null  object 
 12  Suic

# Real Numerical columns

- Age
  - observation: continuous variable showing respondent ages, no missing values.

- Work Pressure
  - observation: numerical but has missing values, measures stress level at work.

- Job Satisfaction
  - observation:numerical, missing values, reflects satisfaction score.

- Work/Study Hours
   - observation: numerical, likely to show distribution of daily/weekly hours spent.

- Financial Stress
   - observation: numerical, non-missing, quantifies financial burden.

# Discrete Numerical Columns

- id
   - observation: id is purely acts as a unique identifier, not used in statistical/ML analysis.

- Age
   - observation: age will show natural clustering around young adults (students) or working-age groups → useful for demographic segmentation.
- Work/Study Hours
    - observation: reflects daily/weekly effort; analyzing its distribution can highlight overwork or under-study patterns.
                                                                                                                
                          

# Categorical Data Columns

- Name 
- Gender 
- City 
- Working Professional or Student 
- Profession 
- Sleep Duration 
- Dietary Habits 
- Degree 
- Suicidal thoughts 

# Checking for Duplicates

In [45]:
df.duplicated().sum()

0

# Observation:
- As seen there are no duplicates found in the dataset.

# Detecting Outliers
- The IQR(Interquartile Range) method was used to detect outliers in all numerical columns as part of the non-visual analysis.

# Steps:
- Calculate Q1(25th percentile) and Q3 (75th percentile).
- Compute IQR = Q3-Q1.
- Define lower and upper bounds.
- Lower bounds = Q1 - 1.5 x IQR.
- Upeer bounds = Q3 + 1.5 x IQR.
- Any value outside this range is considered an outlier.

In [24]:
num_cols = df.select_dtypes(include=['float64', 'int64']).columns

In [25]:
outliers = {}

In [28]:
outliers = {}
for col in num_cols:
    if df[col].isnull().all():
        continue  # Skip the columns with all missing values
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1
    lower = Q1 - 1.5 * IQR
    upper = Q3 + 1.5 * IQR
    outlier_values = df[(df[col] < lower) | (df[col] > upper)][col]
    if not outlier_values.empty:
        outliers[col] = outlier_values.tolist()

In [29]:
if outliers:
    for col, vals in outliers.items():
        print(f"\nOutliers in column '{col}':\n{vals}")
else:
      print("No outliers found in the numerical columns.")

No outliers found in the numerical columns.


# Observation:
- Using the IQR method,outliers were detected in the Projected Openings(ZERO) columns.